doc/source/tutorial/stats/sampling_pinv.rst

```diff
@@ -0,0 +1,300 @@
+Some of the most attractive are:
+
+* It preserves the structural properties of the uniform random number sampler.
+* Transforms a uniform random variate :math:`U` one-to-one into non-uniform
+  random variates :math:`X`.
+* Easy and efficient sampling from truncated distributions.
+
+Unfortunately, the PPF is rarely available in closed form or too slow when
+available. For many distributions, the CDF is also not easy to obtain. This
+method addresses both the shortcomings. The user only has to provide the PDF
+and optionally a point near the mode (called "center") together with the size
+of the maximal acceptable error. It uses a combination of an adaptive and a
+simple Gauss-Lobatto quadrature to obtain the CDF and Newton's interpolation
+to obtain the PPF. The method is not exact, as it only produces random
+variates of the approximated distribution. Nevertheless, the maximal tolerated
+approximation error can be set close to the machine precision and its relative
+error is not much larger than :math:`10^{-15}`. The concept of u-error is used
+to measure and control the error. It is defined as:
+
+.. math::
+    \epsilon_{u}(u) = | u - F\left(F^{-1}_{a}(u)\right) |
+
+where :math:`u \in (0, 1)` is a quantile where we want to measure the error
+and :math:`F^{-1}_a` is the approximated PPF of the given distribution.
+
+The maximal u-error is the criterion for approximation errors when calculating
+the CDF and PPF numerically. The maximal tolerated u-error of an algorithm is
+called the u-resolution of the algorithm and denoted by :math:`\epsilon_{u}`:
+
+.. math::
+    \sup_{u \in (0,1)} | u - F\left(F^{-1}_{a}(u)\right) | \le \epsilon_{u}
+
+The main advantage of the u-error is that it can be easily computed if the CDF
+is available. We refer to [1]_ for a more detailed discussion.
+
+Also, the method only works for bounded distributions. In case of infinite
+tails, the ends of the tails are cut off such that the area under them is
+less than or equal to :math:`0.05\epsilon_{u}`.
+
+There are some restrictions for the given distribution:
+
+* The support of the distribution (i.e., the region where the PDF is strictly
+  positive) must be connected. In practice this means, that the region where
+  PDF is "not too small" must be connected. Unimodal densities satisfy this
+  condition. If this condition is violated then the domain of the distribution
+  might be truncated.
+* When the PDF is integrated numerically, then the given PDF must be
+  continuous and should be smooth.
+* The PDF must be bounded.
+* The algorithm has problems when the distribution has heavy tails (as then
+  the inverse CDF becomes very steep at 0 or 1) and the requested u-resolution
+  is very small. E.g., the Cauchy distribution is likely to show this problem
+  when the requested u-resolution is less then 1.e-12.
+
+.. warning::
+    This method does not work for densities with constant parts (e.g.
```

Comment:
I think the unbounded arcsine density also leads to a seg fault, right? Should we mention it?

Reply:
> I think the unbounded arcsine density also leads to a seg fault, right?

Tested it locally again and it doesn't fail now. Might be a flaky failure while testing locally. I have removed it failing distributions. Let's see if the CI passes.

Reply:
Interesting, `arcsine` is just `beta(0.5,0.5)`. Maybe the tails at 0 and 1 are cut off in a way that PINV works in that case.